File: broadcastreceiverdemo/src/main/java/com/wuhai/broadcastreceiverdemo/MainActivity.java
package com.wuhai.broadcastreceiverdemo;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {


    public static final String TAG = "BroadcastReceiver";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
}

File: kotlinapp/src/main/java/com/wuhai/demo/kotlinapp/MainActivity.kt
package com.wuhai.demo.kotlinapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), View.OnClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btn01.setOnClickListener(this)
    }

    override fun onClick(v: View?) {
        when(v?.id){
            R.id.btn01 ->   Toast.makeText(this,"你好kotlin",Toast.LENGTH_LONG).show()
        }
    }
}
